prolog.cl

```diff
@@ -841,6 +842,747 @@
+         4
+        '("External functions")
+        
+         '("Security integration"
+           "External tables"
+           "Java User-Defined Functions (UDFs)"
+           )
+)
+
+(fact "For external functions. the remote service"
+      '("must" "doesn't have to")
+      "return one row for each row received."
+      )
+
+(fact "RETURNS NULL ON NULL INPUT"
+      '("will not" "will")
+      "call the User-Defined Function (UDF) if any input is null."
+      )
+
+
+  (multi "A Data Engineer is working on a Snowflake deployment in AWS eu-west-1 (Ireland). The Engineer is planning to load data from staged files into target tables using the COPY INTO command.
+Which sources are valid? (Choose 3)"
+         6
+        '("External stage in an Amazon S3 bucket on AWS eu-central-1 (Frankfurt)"
+          "External stage on GCP us-central1 (Iowa)"
+          "External stage in an Amazon S3 bucket on AWS eu-west-1 (Ireland)"
+          )
+        
+         '("Internal stage on GCP us-central1 (Iowa)"
+           "Internal stage on AWS eu-central-1 (Frankfurt)"
+           "SSD attached to an Amazon EC2 instance on AWS eu-west-1 (Ireland)"
+           )
+)
+
+  (multi "A Data Engineer needs to know the details regarding the micro-partition layout for a table named Invoice using a built-in function.
+Which query will provide this information?"
+         4
+        '("SELECT SYSTEM$CLUSTERING_INFORMATION('Invoice');")
+        
+         '("SELECT $CLUSTERING_INFORMATION('Invoice');"
+           "CALL SYSTEM$CLUSTERING_INFORMATION('Invoice');"
+           "CALL $CLUSTERING_INFORMATION('Invoice');"
+           )
+)
+
+  (multi "Assuming a Data Engineer has all appropriate privileges and context, which statements would be used to assess whether the User-Defined Function (UDF), MYDATABASE.SALES.REVENUE_BY_REGION, exists and is secure? (Choose two.)"
+         5
+        '("SHOW USER FUNCTIONS LIKE 'REVENUE_BY_REGION' IN SCHEMA SALES;"
+          "SELECT IS_SECURE FROM INFORMATION_SCHEMA.FUNCTIONS WHERE FUNCTION_SCHEMA = 'SALES' AND FUNCTION_NAME = 'REVENUE_BY_REGION';")
+        
+         '("SELECT IS_SECURE FROM SNOWFLAKE.INFORMATION_SCHEMA.FUNCTIONS WHERE FUNCTION_SCHEMA = 'SALES' AND FUNCTION_NAME = 'REVENUE_BY_REGION';"
+           "SHOW EXTERNAL FUNCTIONS LIKE 'REVENUE_BY_REGION' IN SCHEMA SALES;"
+           "SHOW SECURE FUNCTIONS LIKE 'REVENUE_BY_REGION' IN SCHEMA SALES;"
+           )
+)
+  
+  (multi "A Data Engineer is writing a Python script using the Snowflake Connector for Python. The Engineer will use the snowflake.connector.connect function to connect to Snowflake.
+The requirements are:
+1. Raise an exception if the specified database, schema, or warehouse does not exist
+2. Improve download performance
+Which parameters of the connect function should be used? (Choose two.)
```

Comment:
Remove the "Choose two" instruction.

How does this multi-line presentation look in terminal?

Reply:
![Uploading Screenshot 2024-05-29 at 2.39.58 PM.png…]()

Reply:
<img width="883" alt="Screenshot 2024-05-29 at 2 39 58 PM" src="https://github.com/jhunt/snowpro-examiner/assets/166835366/90e7b5d1-d2d5-4d8a-afad-2cae3e0469d7">
<img width="883" alt="Screenshot 2024-05-29 at 2 39 58 PM" src="https://github.com/jhunt/snowpro-examiner/assets/166835366/90e7b5d1-d2d5-4d8a-afad-2cae3e0469d7">